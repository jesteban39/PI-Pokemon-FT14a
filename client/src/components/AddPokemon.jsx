import { useState } from "react";
import { STAT_NAMES } from "./index";


export default function AddPokemon() {
  const [name, setName] = useState("");
  const [height, setHeight] = useState("");
  const [weight, setWeight] = useState("");
  const [image, setImage] = useState("");
  const [types, setTypes] = useState([]);
  const [stats, setStats] = useState({});

  const grades = ["rock", "normal", "undefine"];

  function handleName(event) {
    const name = event.target.value.replace(/[^a-z\s]/gi, "");
    setName(name);
  }
  function handleHeight(event) {
    const { value } = event.target;
    setHeight(value);
  }
  function handleWeight(event) {
    const { value } = event.target;
    setWeight(value);
  }
  function handleImage(event) {
    const { value } = event.target;
    setImage(value);
  }
  function handleTypes(event) {
    const { name, checked } = event.target;
    console.log("check: ", checked);
    if (checked) {
      setTypes((state) => [...state, name]);
    } else {
      setTypes((state) => state.filter((type) => type !== name));
    }
  }

  function handleStats(event) {
    const { name, value } = event.target;
    setStats((state) => {
      state[name] = value;
      return { ...state };
    });
  }

  return (
    <form>
      <h3>Add a new Pokemon</h3>
      <br />
      <label>
        Name :
        <input value={name} onChange={handleName} />
      </label>
      <br />
      <label>
        Height :
        <input type="number" value={height} onChange={handleHeight} />
      </label>
      <br />
      <label>
        Weight :
        <input type="number" value={weight} onChange={handleWeight} />
      </label>
      <br />
      <label>
        Image :
        <input type="url" value={image} onChange={handleImage} />
      </label>
      <br />
      <label>{types}</label>
      <br />
      {grades.map((type, idx) => (
        <label key={idx}>
          {type}
          <input name={type} type="checkbox" onChange={handleTypes} />
        </label>
      ))}
      <br />
      {STAT_NAMES.map((stat, idx) => (
        <label key={idx}>
          {stat}
          <input
            name={stat}
            type="number"
            value={stats[stat] || ""}
            onChange={handleStats}
          />
        </label>
      ))}
      <br />
      <input type="submit" value="Add" disabled={false} />
    </form>
  );
}
