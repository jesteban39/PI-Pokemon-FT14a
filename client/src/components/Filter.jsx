import { useDispatch, useSelector } from "react-redux";
import "./styles/select.css";
import { ORIGIN_NAMES, Select } from "./index";

export default function Filter() {
  const dispatch = useDispatch();
  const input = useSelector((state) => state.filter);
  const typeNames = [
    "all",
    ...useSelector((state) => state.typeNames),
  ];

  function handleInput(event) {
    const { name, value } = event.target;
    dispatch({
      type: "FILTER",
      payload: { ...input, [name]: value },
    });
    dispatch({ type: "UPDATE_PAGES" });
  }
  return (
    <div className="container-selects">
      <label>{"Filter:"}</label>

      <Select
        cn="select"
        name="origin"
        options={ORIGIN_NAMES}
        onChange={handleInput}
      />
      <Select
        cn="select"
        name="type"
        options={typeNames}
        onChange={handleInput}
      />
    </div>
  );
}
