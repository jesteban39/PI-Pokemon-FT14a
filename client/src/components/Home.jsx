import { useState } from "react";
import { useDispatch,  } from "react-redux";
import { getDetails,  } from "../actions";
import { Link } from "react-router-dom";
import "./styles/home.css";
import {
  Filter,
  Sort,
  PagesPokemons,
} from "./index";

export default function Home() {
  const [name, setName] = useState("");
  const dispatch = useDispatch();

  function handleName(event) {
    let { value } = event.target;
    value = value.toLowerCase().replace(/[^a-z\s]/g, "");
    setName(value);
  }

  function handleSearch(event) {
    dispatch(getDetails(name));
  }

  return (
    <div className="home">
      <section className="home-menu">
        <form >
          <input
            placeholder="Pokemon Name"
            onChange={handleName}
            value={name}
          />
          <Link onClick={handleSearch} to={`/pokemon/${name}`}>
            <input
              className="button-add-home"
              type="submit"
              value="Search"
            />
          </Link>
        </form>

        <Link to="/add">
          <button className="button-add-home">Add pokemon</button>
        </Link>

        <Filter />

        <Sort />
      </section>
      <PagesPokemons />
    </div>
  );
}
