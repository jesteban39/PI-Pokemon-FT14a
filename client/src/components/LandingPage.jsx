import { Link } from "react-router-dom";
import "./styles/landingPage.css";
export default function LandingPage() {
  return (
    <div className="landingPage">
      <p className="welcome">Welcome, on this website you will find a list of all existing pokemons, you can see any of them in detail, you can also filter and order them according to the criteria you choose and you can even create your own custom pokemons.</p>
      <Link to="/home">
        <button className="landingButton">To List Pokemons</button>
      </Link>
    </div>
  );
}
