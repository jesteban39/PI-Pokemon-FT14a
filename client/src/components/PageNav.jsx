export default function (props) {
  const { totalPages, value, onChange } = props;
  return (
    <nav aria-label="Countries Pagination">
      <button
        onClick={() => {
          onChange(1);
        }}
      >
        {"1"}
      </button>
      <button
        onClick={() => {
          if(value > 1) onChange(value - 1);
        }}
      >
        {"<<"}
      </button>
      <button>{value}</button>
      <button
        onClick={() => {
          if(value < totalPages) onChange(value + 1);
        }}
      >
        {">>"}
      </button>
      <button
        onClick={() => {
          onChange(totalPages);
        }}
      >
        {totalPages}
      </button>
    </nav>
  );
}
