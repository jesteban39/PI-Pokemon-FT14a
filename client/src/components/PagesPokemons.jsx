import { useDispatch, useSelector } from "react-redux";
import { useState, useEffect } from "react";
import { PageNav } from "./index";
import { udatePages, fillTypes, getDetails, fillAll } from "../actions";
import { Link } from "react-router-dom";

export default function Pagespages() {
  const state = useSelector((state) => state);
  const dispatch = useDispatch();
  dispatch(fillAll(state));
  dispatch(fillTypes(state));
  const { pages, currentPage } = useSelector((state) => state);

  function handleDetail(event) {
    let id = event.target.name;
    dispatch(getDetails(id));
  }

  function handlePage(value) {
    dispatch({ type: "CURRETN_PAGE", payload: value });
  }

  return (
    <div>
      <PageNav
        totalPages={pages.length}
        value={currentPage}
        onChange={handlePage}
      />

      <span >40 pages Pokemons</span>

      {pages[currentPage - 1].map((pokemon) => (
        <div key={pokemon.id}>
          <Link onClick={handleDetail} to={`/pokemon/:${pokemon.id}`}>
            <img
              name={pokemon.id}
              width="120"
              height="150"
              src={pokemon.img}
              alt="img"
            />
          </Link>

          <div>
            <label>{pokemon.name}</label>
            {pokemon.types.map((type) => (
              <label key={type}>{`  ${type}  `}</label>
            ))}
          </div>
        </div>
      ))}
    </div>
  );
}
