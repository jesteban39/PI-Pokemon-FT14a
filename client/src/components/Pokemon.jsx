import { Link } from "react-router-dom";
export default function Pokemon() {
  let pokemon = {
    id: 30003,
    name: "poke-pru",
    height: 4,
    weight: 60,
    stats: {
      life: 35,
      force: 55,
      defense: 46,
      speed: 44,
    },
    img: "https://assets.pokemon.com/assets/cms2/img/pokedex/full/005.png",
    types: [
      { id: 9, name: "electric" },
      { id: 13, name: "rock" },
    ],
  };
  pokemon.stats = Object.entries(pokemon.stats);
  return (
    <div className="pokemon">
      <br />
      <span>{"pokemon detail"}</span>
      <br />
      <label>{` No.Â° ${pokemon.id} `}</label>
      <img src={pokemon.img} alt="img" />
      <label>{pokemon.name}</label>
      <label>{` Height: ${pokemon.height} `}</label>
      <label>{` Weight: ${pokemon.weight} `}</label>
      <div>
        <label>Types</label>
        <div>
          {pokemon.types.map((type) => (
            <label key={type.id}>{type.name}</label>
          ))}
        </div>
      </div>
      <div>
        <label>Stats</label>
        <div>
          {pokemon.stats.map((stat, idx) => (
            <label key={idx}>{` ${stat[0]}: ${stat[1]} `}</label>
          ))}
        </div>
      </div>
      <Link to="/home">
        <button type="button">Home</button>
      </Link>
    </div>
  );
}
