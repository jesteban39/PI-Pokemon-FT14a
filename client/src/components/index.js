import LandingPage from "./LandingPage.jsx";
import PokemonDetails from "./PokemonDetails";
import Home from "./Home";
import AddPokemon from "./AddPokemon";
import PagesPokemons from "./PagesPokemons";

export const DEFAUL_IMG =
  "https://www.kindpng.com/picc/m/20-202226_yvr-pokeball-poke-ball-hd-png-download.png";
export const STAT_NAMES = ["life", "force", "defense", "speed"];
export const ORIGIN_NAMES = [
  "all",
  "favorite",
  "existing",
  "creations",
];
export const SORT_NAMES = ["number", "name", "force", "life"];

export {
  LandingPage,
  Home,
  PokemonDetails,
  AddPokemon,
  PagesPokemons,
};
