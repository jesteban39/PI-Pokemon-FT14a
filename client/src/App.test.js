import { render, screen } from "@testing-library/react";
import { LandingPage } from "./components";
import { configure, shallow, mount } from "enzyme";

import "@testing-library/jest-dom";

describe("componente <LandingPage />", () => {
  let wrapper = shallow(<LandingPage />);
  beforeEach(() => {
    wrapper = shallow(<LandingPage />);
  });
  test("debería mostrar <LandingPage /> correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debería mostrar un elemento button con el texto To List Pokemons", () => {
    const wrapper = shallow(<LandingPage />);
    const counterText = wrapper.find("button").text().trim();
    expect(counterText).toBe("To List Pokemons");
  });
  test("debería mostrar un parrafo de bienvenida", () => {
    const wrapper = shallow(<LandingPage />);
    const counterText = wrapper.find("p").text().trim();
    expect(/Welcome/i.test(counterText)).toBe.true
  });
});
