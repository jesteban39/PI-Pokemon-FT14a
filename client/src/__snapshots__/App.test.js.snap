// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probamos el componente <LandingPage /> debería mostrar <LandingPage /> correctamente  1`] = `
<div
  className="landingPage"
>
  <p
    className="welcome"
  >
    Welcome, on this website you will find a list of all existing pokemons, you can see any of them in detail, you can also filter and order them according to the criteria you choose and you can even create your own custom pokemons.
  </p>
  <Link
    to="/home"
  >
    <button
      className="landingButton"
    >
      To List Pokemons
    </button>
  </Link>
</div>
`;

exports[`Probamos el componente <LandingPage /> debería mostrar <LandingPage /> correctamente  2`] = `
<div
  className="landingPage"
>
  <p
    className="welcome"
  >
    Welcome, on this website you will find a list of all existing pokemons, you can see any of them in detail, you can also filter and order them according to the criteria you choose and you can even create your own custom pokemons.
  </p>
  <Link
    to="/home"
  >
    <button
      className="landingButton"
    >
      To List Pokemons
    </button>
  </Link>
</div>
`;

exports[`componente <LandingPage /> debería mostrar <LandingPage /> correctamente  1`] = `
<div
  className="landingPage"
>
  <p
    className="welcome"
  >
    Welcome, on this website you will find a list of all existing pokemons, you can see any of them in detail, you can also filter and order them according to the criteria you choose and you can even create your own custom pokemons.
  </p>
  <Link
    to="/home"
  >
    <button
      className="landingButton"
    >
      To List Pokemons
    </button>
  </Link>
</div>
`;
